aws/resource_aws_iam_server_certificate.go

```diff
@@ -140,16 +153,38 @@ func resourceAwsIAMServerCertificateRead(d *schema.ResourceData, meta interface{
 		return fmt.Errorf("error reading IAM Server Certificate (%s): %w", d.Id(), err)
 	}
 
-	d.SetId(aws.StringValue(resp.ServerCertificate.ServerCertificateMetadata.ServerCertificateId))
+	cert := resp.ServerCertificate
+	metadata := cert.ServerCertificateMetadata
+	d.SetId(aws.StringValue(metadata.ServerCertificateId))
 
-	d.Set("certificate_body", resp.ServerCertificate.CertificateBody)
-	d.Set("certificate_chain", resp.ServerCertificate.CertificateChain)
-	d.Set("path", resp.ServerCertificate.ServerCertificateMetadata.Path)
-	d.Set("arn", resp.ServerCertificate.ServerCertificateMetadata.Arn)
+	d.Set("certificate_body", cert.CertificateBody)
+	d.Set("certificate_chain", cert.CertificateChain)
+	d.Set("path", metadata.Path)
+	d.Set("arn", metadata.Arn)
+	d.Set("expiration", aws.TimeValue(metadata.Expiration).Format(time.RFC3339))
+	d.Set("upload_date", aws.TimeValue(metadata.UploadDate).Format(time.RFC3339))
```

Comment:
Nit: Time values may be `nil`, which will return a confusing timestamp from formatting a "0" time. Reference: https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/data-handling-and-conversion.md#root-typestring-and-aws-timestamp

```suggestion
	if metadata.Expiration != nil {
		d.Set("expiration", aws.TimeValue(metadata.Expiration).Format(time.RFC3339))
	} else {
		d.Set("expiration", nil)
	}

	if metadata.UploadDate != nil {
		d.Set("upload_date", aws.TimeValue(metadata.UploadDate).Format(time.RFC3339))
	} else {
		d.Set("upload_date", nil)
	}
```